dracclient/resources/raid.py

```diff
@@ -98,15 +98,295 @@
+
+
 class RAIDManagement(object):
 
+    NAMESPACES = [(uris.DCIM_RAIDEnumeration, RAIDEnumerableAttribute),
+                  (uris.DCIM_RAIDString, RAIDStringAttribute),
+                  (uris.DCIM_RAIDInteger, RAIDIntegerAttribute)]
+
     def __init__(self, client):
         """Creates RAIDManagement object
 
         :param client: an instance of WSManClient
         """
         self.client = client
 
+    def list_raid_controller_settings(self, by_name=False):
+        """List the RAID configuration settings
+
+        :param by_name: Controls whether returned dictionary uses RAID
+                        attribute name or instance_id as key.
+        :returns: a dictionary with the RAID settings using InstanceID as the
+                  key. The attributes are either RAIDEnumerableAttribute,
+                  RAIDStringAttribute objects.
+        :raises: WSManRequestFailure on request failures
+        :raises: WSManInvalidResponse when receiving invalid response
+        :raises: DRACOperationFailed on error reported back by the DRAC
+                interface
+        """
+
+        return utils.list_settings(self.client, self.NAMESPACES,
+                                   by_name=by_name)
+
+    def set_raid_controller_settings(self, new_settings, raid_fqdd):
+        """Sets the RAID configuration
+
+        It sets the pending_value parameter for each of the attributes
+        passed in. For the values to be applied, a config job must
+        be created.
+        :param new_settings: a dictionary containing the proposed values, with
+                         each key being the name of attribute and the value
+                         being the proposed value.
+        :param raid_fqdd: the FQDD of the RAID controller.
+        :returns: a dictionary containing:
+                 - The is_commit_required key with a boolean value indicating
+                   whether a config job must be created for the values to be
+                   applied.
+                 - The is_reboot_required key with a RebootRequired enumerated
+                   value indicating whether the server must be rebooted for the
+                   values to be applied. Possible values are true and false.
+        :raises: WSManRequestFailure on request failures
+        :raises: WSManInvalidResponse when receiving invalid response
+        :raises: DRACOperationFailed on error reported back by the DRAC
+                interface
+        """
+
+        return utils.set_settings('RAID',
+                                  self.client,
+                                  self.NAMESPACES,
+                                  new_settings,
+                                  uris.DCIM_RAIDService,
+                                  "DCIM_RAIDService",
+                                  "DCIM:RAIDService",
+                                  raid_fqdd,
+                                  by_name=False)
+
+    def get_raid_controller_mode(self):
```

Comment:
We should pass in raid_fqdd here in case there is more than 1 RAID controller in the server